tst/EnergyPlus/unit/HVACDXSystem.unit.cc

```diff
@@ -801,4 +802,216 @@ TEST_F(EnergyPlusFixture, VariableSpeedCoils_LatentDegradation_Test)
+
+    bool FirstHVACIteration = true;
+    bool HXUnitOn = false;
+    int InletNode = 1;
+    int ControlNode = 2; // same as outlet node number
+    int condensserNode = 3;
+    int airLoopNum = 1;
+
+    bool zoneEquipment = true;
+    UnitarySystems::UnitarySys::factory(*state, DataHVACGlobals::UnitarySys_AnyCoilType, compName, zoneEquipment, 0);
+    UnitarySystems::UnitarySys *thisSys = &state->dataUnitarySystems->unitarySys[0];
+    // call again to get the rest of the input when sysNum > -1
+    UnitarySystems::UnitarySys::getUnitarySystemInput(*state, compName, false, 0);
+
+    EXPECT_EQ(thisSys->Name, "DX COOLING COIL SYSTEM");
+    EXPECT_FALSE(thisSys->m_ISHundredPercentDOASDXCoil);
+    EXPECT_EQ(thisSys->UnitType, "CoilSystem:Cooling:DX");
+    EXPECT_EQ(thisSys->m_CoolingCoilType_Num, DataHVACGlobals::CoilDX_Cooling);
+    EXPECT_EQ(2, thisSys->m_SystemCoolControlNodeNum);
+
+    // set up outdoor environment
+    state->dataEnvrn->OutDryBulbTemp = 35.0;
+    state->dataEnvrn->OutHumRat = 0.0196;
+    state->dataEnvrn->OutBaroPress = 101325.0;
+    state->dataEnvrn->OutWetBulbTemp = 27.0932;
+    state->dataLoopNodes->Node(condensserNode).Temp = state->dataEnvrn->OutDryBulbTemp;
+    state->dataLoopNodes->Node(condensserNode).HumRat = state->dataEnvrn->OutHumRat;
+
+    // set up inputs to test coil control
+    thisSys->m_DesiredOutletTemp = 20.0;
+    thisSys->m_DesiredOutletHumRat = 1.0;
+    state->dataEnvrn->StdRhoAir = 1.2;
+    state->dataLoopNodes->Node(InletNode).MassFlowRate = 0.8 * state->dataEnvrn->StdRhoAir;
+    state->dataLoopNodes->Node(InletNode).Temp = 24.0;
+    state->dataLoopNodes->Node(InletNode).HumRat = 0.012143698;
+    state->dataLoopNodes->Node(InletNode).Enthalpy = 55029.3778; // conditions at 65 % RH
+    state->dataLoopNodes->Node(ControlNode).TempSetPoint = thisSys->m_DesiredOutletTemp;
+    Real64 RHControlHumRat = 0.01; // humrat at 24C, 60% RH
+    state->dataLoopNodes->Node(ControlNode).HumRatMax = RHControlHumRat;
+    state->dataLoopNodes->Node(ControlNode).HumRat = 0.008;
+
+    // test sensible control
+    int CompOn = 1;
+    state->dataGlobal->BeginEnvrnFlag = true;
+    state->dataSize->CurSysNum = 1;
+    state->dataSize->UnitarySysEqSizing.allocate(1);
+    // run init to size system
+    thisSys->initUnitarySystems(*state, 1, FirstHVACIteration, 0, 0.0);
+    thisSys->controlCoolingSystemToSP(*state, airLoopNum, FirstHVACIteration, HXUnitOn, CompOn);
+    // system meets temperature set point
+    Real64 outTemp1 = state->dataLoopNodes->Node(ControlNode).Temp;
+    Real64 outHumRat1 = state->dataLoopNodes->Node(ControlNode).HumRat;
+    EXPECT_NEAR(thisSys->m_DesiredOutletTemp, state->dataLoopNodes->Node(ControlNode).Temp, 0.001);
+    // system was not told to meet humidity ratio set point (since DesiredOutletHumRat = 1.0)
+    EXPECT_GT(state->dataLoopNodes->Node(ControlNode).HumRat, state->dataLoopNodes->Node(ControlNode).HumRatMax);
+    // sensible load met by compressor speed 3
```

Comment:
note this comment is wrong, it is using compressor speed 1 (verified in my previous testing today).